front/app/containers/Admin/projects/project/timeline/edit.tsx

```diff
@@ -62,79 +60,34 @@ interface IParams {
   id: string | null;
 }
 
-interface DataProps {
-  phases: GetPhasesChildProps;
-}
-
-interface InputProps {}
-
-interface Props extends DataProps, InputProps {}
-
-interface State {
-  locale: Locale | null;
-  phase: IPhase | null;
-  presentationMode: 'map' | 'card';
-  attributeDiff: IUpdatedPhaseProperties;
-  errors: { [fieldName: string]: CLError[] } | null;
-  processing: boolean;
-  saved: boolean;
-  loaded: boolean;
-  inStatePhaseFiles: FileType[];
-  phaseFilesToRemove: FileType[];
-  submitState: 'disabled' | 'enabled' | 'error' | 'success';
-}
-
-class AdminProjectTimelineEdit extends PureComponent<
-  Props & WrappedComponentProps & WithRouterProps,
-  State
-> {
-  params$: BehaviorSubject<IParams | null>;
-  subscriptions: Subscription[];
-
-  constructor(props: Props & WrappedComponentProps & WithRouterProps) {
-    super(props);
-    this.state = {
-      locale: null,
-      phase: null,
-      presentationMode: 'card',
-      attributeDiff: {},
-      errors: null,
-      processing: false,
-      saved: false,
-      loaded: false,
-      inStatePhaseFiles: [],
-      phaseFilesToRemove: [],
-      submitState: 'disabled',
-    };
-    this.subscriptions = [];
-    this.params$ = new BehaviorSubject(null);
-  }
-
-  componentDidMount() {
-    const { projectId, id } = this.props.params;
-
-    this.params$.next({ projectId, id });
-
-    this.subscriptions = [
-      this.params$
-        .pipe(
-          distinctUntilChanged(shallowCompare),
-          switchMap((params: IParams) => {
-            const { id } = params;
-            const locale$ = localeStream().observable;
-            const phase$ = id ? phaseStream(id).observable : of(null);
-            return combineLatest([locale$, phase$]);
-          })
-        )
-        .subscribe(([locale, phase]) => {
-          this.setState({
-            locale,
-            phase,
-            loaded: true,
-          });
-        }),
+type SubmitStateType = 'disabled' | 'enabled' | 'error' | 'success';
 
-      this.params$
+const AdminProjectTimelineEdit = () => {
+  const { projectId, id: phaseId } = useParams() as {
+    projectId: string;
+    id?: string;
+  };
+  const phase = usePhase(phaseId || null);
+  const phases = usePhases(projectId);
+  const [errors, setErrors] = useState<{
+    [fieldName: string]: CLError[];
+  } | null>(null);
+  const [processing, setProcessing] = useState<boolean>(false);
+  const [inStatePhaseFiles, setInStatePhaseFiles] = useState<FileType[]>([]);
+  const [phaseFilesToRemove, setPhaseFilesToRemove] = useState<FileType[]>([]);
+  const [submitState, setSubmitState] = useState<SubmitStateType>('disabled');
+  const [subscriptions, setSubscriptions] = useState<Subscription[]>([]);
+  const [attributeDiff, setAttributeDiff] = useState<IUpdatedPhaseProperties>(
+    {}
+  );
+
+  const params$ = new BehaviorSubject<IParams | null>(null);
```

Comment:
We should get rid of this BehaviourSubject - no need for it, the component will re-render when the params that come from the router change in the useEffect. Just take it from the `useParams` hook above.